clang/lib/ASTMatchers/ASTMatchFinder.cpp

```diff
@@ -1573,19 +1574,42 @@ bool MatchASTVisitor::TraverseAttr(Attr *AttrNode) {
 class MatchASTConsumer : public ASTConsumer {
 public:
   MatchASTConsumer(MatchFinder *Finder,
-                   MatchFinder::ParsingDoneTestCallback *ParsingDone)
-      : Finder(Finder), ParsingDone(ParsingDone) {}
+                   MatchFinder::ParsingDoneTestCallback *ParsingDone,
+                   bool SkipSystemHeaders = false)
```

Comment:
I wonder if it would be better to pass the options instead of a single bool. If we want to introduce more features in the future like only processing the main file and skipping the headers that would avoid having to add more parameters here. We just expand the options.

Reply:
Makes, sense fixed!